### PR TITLE
Upload latest pb ciac 5787

### PR DESCRIPTION
**Pull request by thefrieddan1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5787

## Description
Go over the versions of playbooks. upload only latest playbook.

### CODE REVIEW

**Review by DeanArbel — APPROVED** [review]
Looks good to me.
@yaakovpraisler please CR this as well.

**Review by yaakovpraisler — COMMENTED** [review]
Looks good!

Let's move the logic to a seperate function that will return a boolean on whether to add this content item to the list or not

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -2126,217 +2179,266 @@ def collect_content_items(self):
...
+                        continue
+                    if replace_content_item:
+                        folder_collected_items = [metadata_output
+                                                  if d["id"] == metadata_output["id"]
+                                                  else
+                                                  d for d in folder_collected_items]
+                    if latest_toversion == '':
+                        continue
+                    else:
+                        folder_collected_items.append(metadata_output)
```
Comment (yaakovpraisler) [inline Tests/Marketplace/marketplace_services.py:2441]: Why are we adding the metadata only if `latest_toversion != ''`?

**Review by yaakovpraisler — CHANGES_REQUESTED** [review]
Let's run test-upload with override (`-o`) after fixing the comments to ensure it doesn't break backwards

File: Tests/Marketplace/marketplace_services.py (hunk 1 above)
Comment (thefrieddan1) [reply Tests/Marketplace/marketplace_services.py:2441]: latest_toversion refers to the latest toversion saved to the map and list under that content item id.
if we only got the empty toversion it means we got the latest content item / the only content item and there is no need to add another metadata for that content item.

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -2005,6 +2006,46 @@ def create_local_changelog(self, build_index_folder_path):
...
+        content_item_fromversion = content_item.get('fromversion', '') or content_item.get('fromVersion', '') or ''
+        content_item_toversion = content_item.get(
+            'toversion', MAX_TOVERSION) or content_item.get('toVersion', MAX_TOVERSION) or ''
+        content_item_id = content_item.get('id', '')
+        content_item_latest_version = content_items_to_version_map.setdefault(
+            content_item_id,
+            {'fromversion': content_item_fromversion,
+             'toversion': content_item_toversion
+             })
+        if (replace_old_playbook := content_item_latest_version.get('toversion') < content_item_fromversion):
```
[the remark this answers is not in the document]
Comment (thefrieddan1) [reply Tests/Marketplace/marketplace_services.py:2027]: I tested a case where fromversion and toversion in PB are empty.
Then fromversion get the default value of '' toversion get default value of '99.99.99'
And it gets false in the 2027 if statement which is OK.

**Review by yaakovpraisler — APPROVED** [review]
Good job!

**Pull request merged**